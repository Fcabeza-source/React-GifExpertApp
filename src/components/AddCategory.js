import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState("Ingrese la serie busqueda...");

    const handleInputChange = (e) => {
        setInputValue(e.target.value); 
    };

    const handleSubmit = (e) => {
        e.preventDefault();

        if ( inputValue.trim().length > 2){
            setCategories( cats => [inputValue, ...cats ]);
            setInputValue("");
        }
    };

    const handleErrase = (e) => {
            setInputValue(e.value="");
        }


    return (

        <form onSubmit={handleSubmit}>
            <input 
                type="text"
                value={inputValue}
                onChange={handleInputChange}
                onClick={handleErrase}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories : PropTypes.func.isRequired ,
}
