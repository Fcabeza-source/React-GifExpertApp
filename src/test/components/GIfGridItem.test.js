import "@testing-library/jest-dom";
import React from 'react';
import { shallow } from "enzyme";

import { GifGridItem } from "../../components/GifGridItem";



describe('Pruebas en GifGridGItem.js', () => {


    const title = "Un titulo";
    const url = "https://fnnafonasnfosa/algo.jpg"
    const wrapper = shallow(<GifGridItem title={title} url={url} />);


    test('Debe de mostrar el componente correctamente', () => {

        expect(wrapper).toMatchSnapshot();

    })

    test('Debe de tener un parrafo con title', () => {

        const  p = wrapper.find("p");
        expect( p.text().trim() ).toBe( title );
        
    })
    
    test('Debe de obtener la imagen ', () => {
        
        const img = wrapper.find("img");
        expect( img.prop("src")).toBe(url);
        expect( img.prop("alt")).toBe(title);

    })

    test('Debe tener animate_fadeIn ', () => {

        const div = wrapper.find("div");
        expect (div.prop("className")).toContain("animate__fadeIn");

    })
    
    




})
