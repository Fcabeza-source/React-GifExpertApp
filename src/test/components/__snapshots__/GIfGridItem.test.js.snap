// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridGItem.js Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://fnnafonasnfosa/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
